solr/core/src/java/org/apache/solr/search/BitDocSet.java

```diff
@@ -237,68 +235,15 @@ public BitDocSet clone() {
     return new BitDocSet(bits.clone(), size);
   }
 
-  @Override
-  public Bits getBits(LeafReaderContext context) {
-    if (context.isTopLevel) {
-      return bits;
-    }
-
-    final int base = context.docBase;
-    final int length = context.reader().maxDoc();
-    final FixedBitSet bs = bits;
-
-    return new Bits() {
-      @Override
-      public boolean get(int index) {
-        return bs.get(index + base);
-      }
-
-      @Override
-      public int length() {
-        return length;
-      }
-    };
-  }
-
-  private static final int NO_DOCS_THIS_SEGMENT = -1;
-  private int[] cachedFloorDocs;
-
-  /**
-   * Because `bits.nextSetBit(int)` (called in `nextDoc()`) has no upper limit, lazily cache
-   * floorDocs for each segment to avoid duplicate scanning of bits (and to enable optimization
-   * in consumers afforded by returning <code>null</code> when there are no docs for a given
-   * segment).
-   */
   private int getFloorDoc(final LeafReaderContext ctx) {
```

Comment:
okay but I still don't think you need this method (getFloorDoc) at all.

Reply:
Ah, I wasn't sure whether you were taking issue with the "caching" aspect of this optimization, or the entire idea of front-loading the first "seek" on the bits.